scripts/changesummary.py

```diff
@@ -0,0 +1,540 @@
+        docs_diff["Name"] = api_version_string[0]
+        docs_diff["Version"] = ".".join(api_version_string[1:])
+
+        # These conditions are used as arguments in the `np.where` function
+        # below.
+        deleted_condition = docs_diff["NewValue"].isnull()
+        added_condition = docs_diff["CurrentValue"].isnull()
+
+        # Create a new `ChangeType` column. The `np.where()` function is like a
+        # tenary operator. When the `deleted_condition` is `True`, the
+        # `ChangeType` will be `ChangeType.Deleted`. If the added_condition is
+        # `True` the `ChangeType` will be `ChangeType.Added`, otherwise the
+        # `ChangeType` will be `ChangeType.Changed`.
+        docs_diff["ChangeType"] = np.where(
+            deleted_condition,
+            ChangeType.DELETED,
+            np.where(added_condition, ChangeType.ADDED, ChangeType.CHANGED),
+        )
+
+        # Filter out keys which appear to be less important such as:
+        # {"description", "documentation", "enum", "etag", "revision", "title",
+        # "url", "rootUrl"}
+        docs_diff = docs_diff[
+            ~docs_diff["Key"].str.contains(
+                "|".join(self._get_keys_to_ignore()), case=False
+            )
+        ]
+
+        # Group keys with similar parents together and create a new column
+        # called 'Count' which indicates the number of keys that have been
+        # grouped together. The reason for the count column is that when keys
+        # have the same parent, we group them together to improve readability.
+        docs_diff = (
+            docs_diff.groupby(
+                ["Parent", "Added", "Deleted", "Name", "Version", "ChangeType"]
+            )
+            .size()
+            .reset_index(name="Count")[
+                ["Parent", "Added", "Deleted", "Name", "Version", "ChangeType", "Count"]
+            ]
+        )
+
+        # Rename the Parent column to the Key Column since we are reporting
+        # summary information of keys with the same parent.
+        docs_diff.rename(columns={"Parent": "Key"}, inplace=True)
+        return docs_diff
+
+    def _build_summary_message(self, api_name, is_feature):
+        """Returns a string containing the summary for a given api. The string
+        returned will be in the format `fix(<api_name>): update the API`
+        when `is_feature=False` and `feat(<api_name>)!: update the API`
+        when `is_feature=True`.
+
+        args:
+            api_name (str): The name of the api to include in the summary.
+            is_feature (bool): If True, include the prefix `feat` otherwise use
+                `fix`
+        """
+
+        # Build the conventional commit string based on the arguments provided
+        commit_type = "feat" if is_feature else "fix"
+        return "{0}({1}): update the api".format(commit_type, api_name)
+
+    def _get_keys_to_ignore(self):
+        """Returns keys to ignore as an array of strings. # Filter out keys
```

Comment:
Nit: formatting. Also same comment as above that we should be more specific about why we're ignoring these.

Reply:
Done. https://github.com/googleapis/google-api-python-client/pull/1286/commits/591ac08c7d862010f3823ecd9317baf8635b5f83